### PR TITLE
fix(vue-model-api): do not evict ReactiveINodeJS from cache when objects manged by it are still used

### PR DESCRIPTION
**Pull request by odzhychko**
Garbage collection broke the `vue-model-api` as shown the added test.
Correctly using Vue refs fixes it.

## To be verified by reviewers

* [ ] Relevant public API members have been documented
* [ ] Documentation related to this PR is complete
  * [ ] Boundary conditions are documented
  * [ ] Exceptions are documented
  * [ ] Nullability is documented if used
* [ ] Touched existing code has been extended with documentation if missing
* [ ] Code is readable
* [ ] New features and fixed bugs are covered by tests

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -50,7 +50,7 @@ kotlin-stdlib-common = { group = "org.jetbrains.kotlin", name = "kotlin-stdlib-c
 kotlin-stdlib-jdk8 = { group = "org.jetbrains.kotlin", name = "kotlin-stdlib-jdk8", version.ref = "kotlin" }
 kotlin-reflect = { group = "org.jetbrains.kotlin", name = "kotlin-reflect", version.ref = "kotlin" }
 kotlin-serialization-json = { group = "org.jetbrains.kotlinx", name = "kotlinx-serialization-json", version.ref = "kotlinSerialization" }
-kotlin-serialization-yaml = { group = "com.charleskorn.kaml", name = "kaml", version = "0.62.2" }
+kotlin-serialization-yaml = { group = "com.charleskorn.kaml", name = "kaml", version = "0.63.0" }
```
Comment (odzhychko) [inline gradle/libs.versions.toml:53]: That commit should not be part of this PR.

File: vue-model-api/src/internal/runGarbageCollection.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  if (global.gc) {
+    global.gc();
+  } else {
+    fail(
+      "Garbage collection unavailable. Pass --expose-gc when launching node to enable forced garbage collection.",
+    );
+  }
+}
+
+test("test garbage collection invocation", async () => {
```
Comment (odzhychko) [inline vue-model-api/src/internal/runGarbageCollection.ts:18]: Should not be here.

**Review by mhuster23 — APPROVED** [review]
